### PR TITLE
Update Security section in Istio overview

### PR DESCRIPTION
**Pull request by wattli**

### CODE REVIEW

File: _docs/concepts/what-is-istio/overview.md [hunk 1]
```diff
@@ -97,13 +97,13 @@ This loose coupling allows Istio to run on multiple environments
...
 
-[Istio-Auth]({{home}}/docs/concepts/security/mutual-tls.html) provides strong service-to-service and end-user authentication using mutual TLS, with built-in identity and credential management.
-It can be used to upgrade unencrypted traffic in the service mesh, and provides operators the ability to enforce policy based
-on service identity rather than network controls. Future releases of Istio will add fine-grained access control and auditing to control
-and monitor who accesses your service, API, or resource, using a variety of access control mechanisms, including attribute and
-role-based access control as well as authorization hooks.
+[Security]({{home}}/docs/concepts/security/) provides strong service-to-service and end-user authentication, with built-in identity and
+credential management. It can be used to upgrade unencrypted traffic in the service mesh, and provides operators the ability to enforce
+policy based on service identity rather than network controls. Starting from release 0.5, Istio supports
+[role-based access control]({{home}}/docs/concepts/security/rbac.html) to control who can access your services, API, or resource. Future
```
Comment (liminw) [inline _docs/concepts/what-is-istio/overview.md]: "control who can access your services". Remove "API" (same as "service", redundant) and "resource".

Comment (wattli) [reply _docs/concepts/what-is-istio/overview.md]: Done.

File: _docs/concepts/what-is-istio/overview.md [hunk 2]
```diff
@@ -97,13 +97,13 @@ This loose coupling allows Istio to run on multiple environments
 (e.g., Kubernetes, Consul/Nomad) while maintaining the same operator
 interface for traffic management.
 
-### Istio-Auth
+### Security
```
Comment (geeknoid) [inline _docs/concepts/what-is-istio/overview.md]: I don't think we want to use "Security" here. The list is about individual components "Mixer", "Pilot", "Envoy". Security is not a proper noun.

This kind of situation is why I was mentioning that we really should name Istio Auth or Istio CA something specific.

File: _docs/concepts/what-is-istio/overview.md [hunk 3]
```diff
@@ -97,13 +97,13 @@ This loose coupling allows Istio to run on multiple environments
...
 
-[Istio-Auth]({{home}}/docs/concepts/security/mutual-tls.html) provides strong service-to-service and end-user authentication using mutual TLS, with built-in identity and credential management.
-It can be used to upgrade unencrypted traffic in the service mesh, and provides operators the ability to enforce policy based
-on service identity rather than network controls. Future releases of Istio will add fine-grained access control and auditing to control
-and monitor who accesses your service, API, or resource, using a variety of access control mechanisms, including attribute and
-role-based access control as well as authorization hooks.
+[Security]({{home}}/docs/concepts/security/) provides strong service-to-service and end-user authentication, with built-in identity and
+credential management. It can be used to upgrade unencrypted traffic in the service mesh, and provides operators the ability to enforce
+policy based on service identity rather than network controls. Starting from release 0.5, Istio supports
```
Comment (geeknoid) [inline _docs/concepts/what-is-istio/overview.md]: I think we should remove the "starting from release 0.5" part. This is 0.8 docs and in general we describe many things that didn't exist in 0.4.

File: _docs/concepts/what-is-istio/overview.md (hunk 2 above)
Comment (geeknoid) [reply _docs/concepts/what-is-istio/overview.md]: Also note that the diagram on this page says Istio-Auth.

**Comment by geeknoid** [comment]
Our site currently contains tons of references to "Istio Auth" and "Istio-Auth". I think we should either revert this change so we're consistent, or change all other references. And as I said, we can't use Security as a proper noun.

So I think we should redo this change with Istio-Auth until we find a suitable replacement and apply it consistently throughout the web site, including in all diagrams.

**Pull request merged**